renga_deployer/engines.py

```diff
@@ -498,20 +522,47 @@ def get_host_ports(self, execution):
             return {'ports': []}
 
         else:
-            return {
-                'ports': [{
-                    'specified':
-                    port.port,
-                    'host':
-                    current_app.config[
-                        'DEPLOYER_K8S_CONTAINER_IP'] or pod.items[
-                            0].status.host_ip,
-                    'exposed':
-                    port.node_port,
-                    'protocol':
-                    port.protocol,
-                } for port in service.items[0].spec.ports]
-            }
+            if current_app.config.get(
+                    'DEPLOYER_K8S_USE_INGRESS'):   # pragma no cover
+
+                beta_api = self._kubernetes.client.ExtensionsV1beta1Api()
+                ingress = beta_api.list_namespaced_ingress(
+                    execution.namespace,
+                    label_selector='job-uid={0}'.format(
+                        execution.engine_id)).items[0]
+
+                return {
+                    'ports': [{
+                        'specified':
+                        port.port,
+                        'host':
+                        current_app.config[
+                            'DEPLOYER_K8S_CONTAINER_IP'] or
+                            ingress.status.load_balancer.ingress[0].ip,
+                        'path':
+                        ingress.spec.rules[0].http.paths[0].path,
+                        'exposed':
+                        '443',
+                        'protocol':
+                        port.protocol,
+                    } for port in service.items[0].spec.ports]
+                }
+
+            else:
```

Comment:
Using `else` & `return` is an anti-pattern in Python. Can you remove the else and remove one level of indentation?